airflow/ui/src/components/TriggerDag/TriggerDAGModal.tsx

```diff
@@ -0,0 +1,99 @@
+/*!
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { Text, Heading, VStack } from "@chakra-ui/react";
+import React, { useCallback, useEffect, useMemo, useState } from "react";
+
+import { Dialog } from "src/components/ui";
+
+import TriggerDAGForm from "./TriggerDAGForm";
+import type { DagParams } from "./TriggerDag";
+import { TriggerDag as triggerDag } from "./TriggerDag";
+
+type TriggerDAGModalProps = {
+  dagDisplayName: string;
+  dagId: string;
+  onClose: () => void;
+  open: boolean;
+};
+
+const TriggerDAGModal: React.FC<TriggerDAGModalProps> = ({
+  dagDisplayName,
+  dagId,
+  onClose,
+  open,
+}) => {
+  const initialDagParams = useMemo(
+    () => ({
+      configJson: "{}",
+      dagId,
+      logicalDate: "",
+      runId: "",
+    }),
+    [dagId],
+  );
+
+  const [dagParams, setDagParams] = useState<DagParams>(initialDagParams);
+
+  const handleTrigger = useCallback(
+    (updatedDagParams: DagParams) => {
+      triggerDag(updatedDagParams);
+      onClose();
+    },
+    [onClose],
+  );
+
+  useEffect(() => {
+    if (!open) {
+      setDagParams(initialDagParams);
+    }
+  }, [open, initialDagParams]);
+
+  return (
+    <Dialog.Root onOpenChange={onClose} open={open} size="xl">
+      <Dialog.Content backdrop>
+        <Dialog.Header>
+          <VStack align="start" gap={2}>
+            <Heading size="xl">
+              Trigger DAG{" "}
+              {dagDisplayName ? <span>- {dagDisplayName}</span> : undefined}
+            </Heading>
+            {!dagDisplayName && (
```

Comment:
Small bug - for me this now _never_ shows up. I assume from the backend the `dagDisplayName` is always filled with DAG ID if not given. So most probably you need to check via `dagDisplayName !== dagId`

Reply:
Ohh yes, thats right. Let me do that.

Reply:
Updated UI: When dag id is not same as display name.
<img width="927" alt="image" src="https://github.com/user-attachments/assets/2d49131c-166d-4680-85c2-c8e0628b38aa">

Reply:
We actually don't use `dag_id` natively in the UI yet. so I'm not sure if this is specifically necessary for triggering a run.

Reply:
It was just an extra information.
Okay, I will remove it.